### PR TITLE
#53 API.Models, API.Context - added properties for Entity, overide for SaveChangesAsync

### PR DESCRIPTION
**Pull request by AntonNorgren**
Updated the entity class with properties.
Added SaveChangesAsync override to update entity version.

### CODE REVIEW

**Review by Gerglamesh — APPROVED** [review]
Ser bra ut!

File: KNet/KNet.API/Models/Entity.cs [hunk 1]
```diff
@@ -2,8 +2,16 @@
...
-    public class Entity
+    public abstract class Entity
     {
+        public Entity()
+        {
+            Created = Modified = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
+        }
         public Guid Id { get; set; }
+        public bool IsActive { get; set; }
+        public int Version { get; set; } = 1;
```
Comment (PontusGillenang) [inline KNet/KNet.API/Models/Entity.cs:13]: I do not really understand this property, is this for identifying the entity version for this object?

Comment (AntonNorgren) [reply KNet/KNet.API/Models/Entity.cs:13]: By setting IsActive to false you can soft-delete entities from the database.
Version is set to 1 by default, and is incremented whenever an entity is modified.

**Review by PontusGillenang — APPROVED** [review]
Well done!

**Pull request merged**